Scripts/script-IndicatorMaliciousRatioCalculation.yml

```diff
@@ -0,0 +1,110 @@
+commonfields:
+  id: IndicatorMaliciousRatioCalculation
+  version: -1
+name: IndicatorMaliciousRatioCalculation
+script: |-
+  import json, hashlib
+
+  APPEARS_IN_MIN_NUMBER_OF_INCIDENTS = int(demisto.args()['appearsInMinNumberOfIncidents'])
+  MAX_INCIDENTS = int(demisto.args()['maxIncidents'])
+  MAX_INDICATORS = int(demisto.args()['maxIndicators'])
+  QUERY = demisto.args()['query']
+
+  def get_incident_labels_map(labels):
+      if labels is None:
+          return {}
+      labels_map = {}
+
+      for label in labels:
+          label_type = label['type'].lower()
+          labels_map[label_type] = label['value']
+      return labels_map
+
+  def hash_object(str_list_dict):
+      if str_list_dict == "" or str_list_dict is None:
+          return str_list_dict
+      if (type(str_list_dict)) == dict:
+          return dict(map(lambda (k, v): (k, hash_object(v)), str_list_dict.iteritems()))
+      if (type(str_list_dict) == list):
+          return map(lambda x: hash_object(x), str_list_dict)
+
+      if (type(str_list_dict) in [str, unicode]):
+          str_value = str_list_dict.encode('utf-8')
+      else:
+          str_value = str(str_list_dict)
+      return hashlib.md5(str_value).hexdigest()
+
+  def get_indiator_data(indicator):
+      return {k: v for k,v in indicator.items() if k in ['indicator_type', 'firstSeen', 'lastSeen', 'investigationIDs', 'manualScore', 'id', 'score', 'value']}
+
+  def get_incident_data(incident):
+      result = {k: v for k,v in incident.items() if k in ['id', 'type', 'occurred', 'CustomFields', 'labels']}
+      result['labels'] = hash_object(get_incident_labels_map(result.get('labels')))
+      result['CustomFields'] = hash_object(result.get('CustomFields'))
+      return result
+
+
+  res = demisto.executeCommand("findIndicators", {'query': QUERY, 'size': MAX_INDICATORS})
+  indicators = res[0]['Contents']
+  indicators_result = map(get_indiator_data, indicators)
+  res = demisto.executeCommand("getIncidents", {'query': QUERY.replace('incident.', ''), 'size': MAX_INCIDENTS})
```

Comment:
better to query all incidents that have the indicators ids found in indicators_result
More efficient and correct

Reply:
I have to do it in this way so I can divide the incidents to resovled and non-resolved. otherwise I will get all the incidents as a result and it will hard to parse it (the fields to identify if it's resolved or not is actually in the query).
Plus I check the efficient - it's really really slow to query by indicator.id - I guess it's cuz it's a long "or" query and it's not efficient